### PR TITLE
[fix] #2384: Replace decode with decode_all.

### PR DESCRIPTION
**Pull request by SamHSmith**
Signed-off-by: Sam H. Smith <sam.henning.smith@protonmail.com>

### CODE REVIEW

**Review by ilchu — CHANGES_REQUESTED** [review]
LGTM, will approve after missing CI fixes.

File: wasm/src/lib.rs [hunk 1]
```diff
@@ -305,7 +305,7 @@ unsafe fn _decode_from_raw_in_range<T: Decode>(
     let bytes = Box::from_raw(core::slice::from_raw_parts_mut(ptr as *mut _, len));
```
Comment (mversic) [inline wasm/src/lib.rs:305]: there are a few places in this module that you should replace `Decode` bound with `DecodeAll` as you did in `smartcontracts/wasm.rs`

File: version/derive/src/lib.rs [hunk 2]
```diff
@@ -242,25 +242,6 @@ impl Parse for DeclareVersionedArgs {
...
-                            *version,
-                            RawVersioned::ScaleBytes(input.to_vec())
-                        ))))
-                    }
-                } else {
-                    Err(Error::NotVersioned)
-                }
-            }
-
             fn decode_all_versioned(input: &[u8]) -> iroha_version::error::Result<Self> {
```
Comment (mversic) [inline version/derive/src/lib.rs:245]: why does the implementation of `decode_all_versioned` use `::decode`?

**Review by mversic — CHANGES_REQUESTED** [review]

File: wasm/src/lib.rs (hunk 1 above)
Comment (mversic) [reply wasm/src/lib.rs:305]: also check if `DumpDecoded` in `tools/parity_scale_codec` should be modified

File: version/derive/src/lib.rs (hunk 2 above)
Comment (SamHSmith) [reply version/derive/src/lib.rs:245]: I thought I wouldn't change the implementation unnecessarily. But I've changed it to use decode_all now.

**Review by mversic — APPROVED** [review]

**Review by ilchu — APPROVED** [review]

**Review by mversic — APPROVED** [review]

**Review by ilchu — APPROVED** [review]

**Pull request merged**